jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -476,7 +480,7 @@ else if (ExceptionUtil.areNotAssociated(_failure.getFailure(), x) && _failure.ge
                     }
 
                     // If the application has not been otherwise informed of the failure
-                    if (invokeOnContentAvailable == null && invokeWriteFailure == null)
+                    if (invokeOnContentAvailable == null && invokeWriteFailure == null && onFailure == null)
```

Comment:
This feels fishy.
Been trying to remind myself of the changes we made in #9684 and when we removed many of these onFailure tests.  Not having much luck.

Reply:
@joakime to me it was stranger that that we would notify listeners AND fail the callback.  Such double actions are frequently wrong.    This feels much better to explain: if we have told the app about the failure, then it is still the apps responsibility to complete the callback, but if we have not told them about it then it is ours.

Reply:
@joakime I refined the logic and comments a little bit. Less fishy?

Reply:
I still have this feeling like we are restoring the behavior from before #9684, but I honestly cannot prove it.
Oh well, all of the tests pass, and we added a bunch more during the #9684 effort, so I'm happy with this as-is.

Reply:
> I still have this feeling like we are restoring the behavior from before #9684

I see it as continuing the work of #9684 (and other recent error changes).
Previously we were doing everything on failure: fail any IO operations, called the listeners & failed the callback. 

We have moved to a mechanism where we almost always only do a single notification... except prior to this PR we could call the error callback AND fail the callback.

So this PR really just competes previous work that was reducing dual notification.

We now only get dual notification if you have both demand and a write outstanding.